src/main/java/seedu/address/model/MasterDeck.java

```diff
@@ -150,6 +151,16 @@ public void setDeck(Deck target, Deck editedDeck) {
      * {@code key} must exist.
      */
     public void removeDeck(Deck key) {
+        ArrayList<Card> remCards = new ArrayList<>();
+        for (Card c : cards) {
+            Deck targetDeck = c.getDeck().isPresent() ? c.getDeck().get() : new Deck("");
```

Comment:
Not very important but you can do this a bit cleaner by calling `targetDeck = c.getDeck().orElse(new Deck)`